src/carousel/carousel.component.html

```diff
@@ -1,13 +1,15 @@
-<div (mouseenter)="pause()" (mouseleave)="play()" (mouseup)="play()" class="carousel slide">
+<div (mouseenter)="pause()" (mouseleave)="play()" (mouseup)="play()" class="carousel slide" (keydown)="keydownPress($event)" (focusin)="pauseFocusIn()" (focusout)="pauseFocusOut()" tabindex="0">
   <ol class="carousel-indicators" *ngIf="showIndicators && slides.length > 1">
     <li *ngFor="let slidez of slides; let i = index;" [class.active]="slidez.active === true" (click)="selectSlide(i)"></li>
   </ol>
   <div class="carousel-inner"><ng-content></ng-content></div>
-  <a class="left carousel-control carousel-control-prev" [class.disabled]="activeSlide === 0 && noWrap" (click)="previousSlide()" *ngIf="slides.length > 1">
+  <a class="left carousel-control carousel-control-prev" [class.disabled]="activeSlide === 0 && noWrap" (click)="previousSlide()" (focus)="prevKey($event)" *ngIf="slides.length > 1"
```

Comment:
`(focus)="prevKey($event)"`
this doesn't make much sense to me, because in `prevKey` you're checking event keycode, but focus event object doesn't even have it, so the code of `prevKey` will never be executed. remove this binding and `prevKey` method as well